xwiki-commons-core/xwiki-commons-displayer/src/main/java/org/xwiki/displayer/HTMLDisplayer.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.displayer;
+
+import java.util.Map;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.stability.Unstable;
+
+/**
+ * Display an html element based on the type of {@link T}.
+ *
+ * @param <T> type used for the display
+ * @version $Id$
+ * @since 10.11RC1
+ */
+@Role
+@Unstable
+public interface HTMLDisplayer<T>
+{
+    /**
+     * @param value the value on which the display is based on
+     * @return the html element based on the value and the type
```

Comment:
```suggestion
     * @return the HTML that can be used to display the given value
```
First of all, the returned string could have multiple HTML "elements" and it could contain comments or other HTML syntax. We could have for instance:

    <input type="checkbox" name="enabled" value="true" />
    <!-- The value submitted when the checkbox above is not clicked -->
    <input type="hidden" name="enabled" value="false" />

Then, from what I know, "HTML element" is used mostly when your talk about **object** representation (e.g. DOM element), while "HTML tag" is used to denote the **string** representation. So you can say "return the HTML tags" or "return the HTML markup" or simply "return the HTML".

Reply:
I agree, "return the HTML markup" or "return the HTML" is more accurate.